projects/client/RabbitMQ.Client/src/client/impl/SocketFrameHandler.cs

```diff
@@ -231,7 +231,7 @@ public void WriteFrame(OutboundFrame frame)
             var nbw = new NetworkBinaryWriter(ms);
             frame.WriteTo(nbw);
             m_socket.Client.Poll(m_writeableStateTimeout, SelectMode.SelectWrite);
-            Write(ms.ToArray());
+            Write(ms.GetBufferSegment());
```

Comment:
Could you tell why do we not write frames to `m_writer` directly instead of buffering them first?

Reply:
@michaelklishin @kjnilsson

Reply:
Thanks 👍 

So, the only way to reduce allocations is to pool `MemoryStream`.